In [39]:
import pandas as pd
pd.set_option('display.max_columns',None)
import warnings
warnings.filterwarnings('ignore')

In [13]:
df = pd.read_csv(r'C:\Users\HP\Desktop\Python\Vs_Python\Data analysis\RCA-Sales-Super-Store\Dataset\Sample - Superstore.csv', encoding='latin1')

In [31]:
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,California,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


In [15]:
df.shape

(9994, 21)

In [16]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

In [18]:
df.isnull().sum()

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64

In [19]:
df.duplicated().sum()

np.int64(0)

In [20]:
df.columns

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit'],
      dtype='object')

In [21]:
orders = df[['Order ID', 'Order Date', 'Ship Date', 'Ship Mode']]
customers = df[['Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State','Postal Code', 'Region']]
products = df[['Product ID', 'Category', 'Sub-Category', 'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit']]

In [33]:
total_sales = df['Sales'].sum()

In [42]:
total_sales

np.float64(2297200.8603000003)

In [34]:
total_profit = df['Profit'].sum()

In [43]:
total_profit

np.float64(286397.0217)

In [35]:
profit_margin = (total_profit / total_sales) * 100

print(f"Profit Margin: {profit_margin:.2f}%")

Profit Margin: 12.47%


In [ ]:
df['Profit_Margin_%'] = (df['Profit'] / df['Sales']) * 100

In [40]:
products['Profit_Margin_%'] = (products['Profit'] / products['Sales']) * 100

,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit,Profit_Margin_%
0,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136,16.00
1,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820,30.00
2,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714,47.00
3,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310,-40.00
4,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164,11.25
...,...,...,...,...,...,...,...,...,...
9989,FUR-FU-10001889,Furniture,Furnishings,Ultra Door Pull Handle,25.2480,3,0.20,4.1028,16.25
9990,FUR-FU-10000747,Furniture,Furnishings,Tenex B1-RE Series Chair Mats for Low Pile Car...,91.9600,2,0.00,15.6332,17.00
9991,TEC-PH-10003645,Technology,Phones,Aastra 57i VoIP phone,258.5760,2,0.20,19.3932,7.50
9992,OFF-PA-10004041,Office Supplies,Paper,"It's Hot Message Books with Stickers, 2 3/4"" x 5""",29.6000,4,0.00,13.3200,45.00


### INSIGHT
    - The business generated ₹22.97 lakh in revenue with a profit margin of 12.47%, yielding approximately ₹2.86 lakh in profit.

In [65]:
products.groupby(['Category'])['Profit'].sum()

Category
Furniture           18451.2728
Office Supplies    122490.8008
Technology         145454.9481
Name: Profit, dtype: float64

- Furniture sale generate toal profit $18451.27 (Lowest)
- Office Supplies sale generate toal profit $122490.80 (Medium)
- Technology sale generate toal profit $145454.94 (Most)

In [66]:
products.groupby(['Category'])['Profit'].mean()

Category
Furniture           8.699327
Office Supplies    20.327050
Technology         78.752002
Name: Profit, dtype: float64

- Each Furniture sale generates about $8.70 profit
- Each Office Supplies sale generates about $20.33 profit
- Each Technology sale generates about $78.75 profit

In [46]:
products.groupby(['Category'])['Profit_Margin_%'].mean()

Category
Furniture           3.878353
Office Supplies    13.803029
Technology         15.613805
Name: Profit_Margin_%, dtype: float64

### INSIGHT
    - Technology Category generate 15% profit margin(Most Profitable Category)
    - Furniture Category generate 3% profit margin(Lowest profit Category)

In [67]:
products[products['Category'] == 'Furniture'].groupby('Sub-Category')['Profit'].sum().sort_values()

Sub-Category
Tables        -17725.4811
Bookcases      -3472.5560
Furnishings    13059.1436
Chairs         26590.1663
Name: Profit, dtype: float64

- In Furniture the Sub-category [Tables, Bookcases] are in Negative Profit

In [68]:
products[products['Category'] == 'Furniture'].groupby('Sub-Category')['Profit_Margin_%'].mean().sort_values()

Sub-Category
Tables        -14.772653
Bookcases     -12.664007
Chairs          4.389963
Furnishings    13.706635
Name: Profit_Margin_%, dtype: float64

- Tables is 14% Negative
- Bookcases is 12% Negative

In [79]:
print(products['Profit_Margin_%'].skew())
print(products['Profit'].skew())
print(products['Sales'].skew())

-2.8948263069056015
7.561431562468343
12.97275234181623


In [84]:
products[products['Sub-Category'] == 'Tables'].groupby('Discount')[['Sales', 'Profit']].sum()

,Sales,Profit
Discount,,
0.00,71578.760,13276.2997
0.20,45430.232,-303.5580
0.30,25182.150,-3402.3276
0.40,45614.406,-16187.3968
0.45,5484.974,-2493.1111
0.50,13675.010,-8615.3873


In [85]:
df[df['Sub-Category'] == 'Tables'].groupby('State')['Profit'].sum().sort_values()

State
New York         -4535.6408
Illinois         -4309.7447
North Carolina   -3684.2539
Ohio             -2715.3345
Tennessee        -2663.4138
Pennsylvania     -2588.7538
Florida          -2493.1111
Arizona          -2281.8701
Texas            -2216.6766
Oregon           -1048.8971
Colorado          -974.8754
Massachusetts     -545.2947
California        -303.5580
Rhode Island      -230.1678
New Hampshire     -105.3164
Delaware           -85.8963
West Virginia      -76.9536
Maryland           -71.1170
New Jersey         -51.2906
Connecticut        -19.6146
Arkansas            45.2940
Iowa               106.6248
Georgia            137.9470
Wisconsin          147.5022
Kentucky           160.1586
Minnesota          270.4382
Utah               358.9224
Missouri           361.5234
Louisiana          368.3944
Mississippi        411.1676
Indiana            412.9218
Idaho              419.7063
Nevada             437.1892
Alabama            556.0730
Oklahoma           579.8830
Michigan      

In [86]:
products[products['Sub-Category'] == 'Tables'].groupby('Product Name')['Profit'].sum().sort_values().head(10)

Product Name
Chromcraft Bull-Nose Wood Oval Conference Tables & Bases                       -2876.1156
Bush Advantage Collection Racetrack Conference Table                           -1934.3976
Balt Solid Wood Round Tables                                                   -1201.0581
BoxOffice By Design Rectangular and Half-Moon Meeting Room Tables              -1148.4375
Riverside Furniture Oval Coffee Table, Oval End Table, End Table with Drawer   -1147.4000
Bretford “Just In Time” Height-Adjustable Multi-Task Work Tables                -964.1940
Bevis Oval Conference Table, Walnut                                             -856.0144
BPI Conference Tables                                                           -795.9725
Hon 94000 Series Round Tables                                                   -681.2140
Chromcraft Bull-Nose Wood 48" x 96" Rectangular Conference Tables               -611.5878
Name: Profit, dtype: float64

In [87]:
df[df['Sub-Category'] == 'Tables'].groupby(['State', 'Discount'])['Profit'].sum().sort_values()

State           Discount
New York        0.40       -4535.6408
Illinois        0.50       -4309.7447
North Carolina  0.40       -3684.2539
Ohio            0.40       -2715.3345
Tennessee       0.40       -2663.4138
Pennsylvania    0.40       -2588.7538
Florida         0.45       -2493.1111
Arizona         0.50       -2281.8701
Texas           0.30       -2216.6766
Oregon          0.50       -1048.8971
Colorado        0.50        -974.8754
Massachusetts   0.30        -545.2947
California      0.20        -303.5580
Rhode Island    0.30        -230.1678
New Hampshire   0.30        -105.3164
Delaware        0.30         -85.8963
West Virginia   0.30         -76.9536
Maryland        0.30         -71.1170
New Jersey      0.30         -51.2906
Connecticut     0.30         -19.6146
Arkansas        0.00          45.2940
Iowa            0.00         106.6248
Georgia         0.00         137.9470
Wisconsin       0.00         147.5022
Kentucky        0.00         160.1586
Minnesota       0.00     

In [88]:
tables = df[df['Sub-Category'] == 'Tables']

state_margin = (
    tables.groupby('State')
    .agg({
        'Sales':'sum',
        'Profit':'sum'
    })
)

state_margin['Profit_Margin_%'] = (
    state_margin['Profit']
    / state_margin['Sales']
) * 100

state_margin.sort_values('Profit_Margin_%')

,Sales,Profit,Profit_Margin_%
State,,,
Colorado,1457.130,-974.8754,-66.903804
Illinois,6550.670,-4309.7447,-65.790899
Oregon,1671.860,-1048.8971,-62.738333
Arizona,3995.350,-2281.8701,-57.113147
Florida,5484.974,-2493.1111,-45.453472
Tennessee,6214.356,-2663.4138,-42.859048
North Carolina,9681.732,-3684.2539,-38.053665
Ohio,7887.114,-2715.3345,-34.427479
New York,13779.018,-4535.6408,-32.917010


In [ ]:
df[df['Sub-Category'] == 'Tables'] \
.groupby('Discount') \
.agg({
    'Sales':'sum',
    'Profit':'sum',
    'Order ID':'count'
})

,Sales,Profit,Order ID
Discount,,,
0.00,71578.760,13276.2997,72
0.20,45430.232,-303.5580,71
0.30,25182.150,-3402.3276,54
0.40,45614.406,-16187.3968,75
0.45,5484.974,-2493.1111,11
0.50,13675.010,-8615.3873,36
